src/General-Rules/ReBaselineWithProperSuperclassRule.class.st

```diff
@@ -0,0 +1,52 @@
+"
+A custom baseline class should be a (direct or indirect) subclass of the predefined class BaselineOf. So use BaselineOf as the superclass for your baseline class.
+"
+Class {
+	#name : 'ReBaselineWithProperSuperclassRule',
+	#superclass : 'ReAbstractRule',
+	#category : 'General-Rules-Migrated',
+	#package : 'General-Rules',
+	#tag : 'Migrated'
+}
+
+{ #category : 'testing - interest' }
+ReBaselineWithProperSuperclassRule class >> checksClass [
+
+	^ true
+]
+
+{ #category : 'manifest' }
+ReBaselineWithProperSuperclassRule class >> uniqueIdentifierName [ 
+	"This number should be unique and should change only when the rule completely change semantics"
+
+	^'ReBaselineWithProperSuperclassRule'
+]
+
+{ #category : 'running' }
+ReBaselineWithProperSuperclassRule >> basicCheck: aClass [
+
+	aClass name = #BaselineOf ifTrue: [ ^ false ].
+	
+	^(aClass name beginsWith: 'BaselineOf') 
+		and: [ (aClass inheritsFrom: BaselineOf) not ]
```

Comment:
I think it should be a direct subclass no? Maybe `aClass superclass = BaselineOf` would be better?

Reply:
@jecisc I already thought about it, yes. It not necessarily must be a direct subclass of BaselineOf.  So I decided to use #inheritsFrom: 

One can have a common baseline superclass in a project - rare and exotic (I know) - but can happen.